opengrok-indexer/src/main/java/org/opengrok/indexer/Metrics.java

```diff
@@ -22,19 +22,105 @@
  */
 package org.opengrok.indexer;
 
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
 import io.micrometer.core.instrument.binder.jvm.ClassLoaderMetrics;
 import io.micrometer.core.instrument.binder.jvm.JvmGcMetrics;
 import io.micrometer.core.instrument.binder.jvm.JvmMemoryMetrics;
 import io.micrometer.core.instrument.binder.jvm.JvmThreadMetrics;
 import io.micrometer.core.instrument.binder.system.ProcessorMetrics;
 import io.micrometer.prometheus.PrometheusConfig;
 import io.micrometer.prometheus.PrometheusMeterRegistry;
+import io.micrometer.statsd.StatsdConfig;
+import io.micrometer.statsd.StatsdMeterRegistry;
+import io.micrometer.statsd.StatsdFlavor;
+import org.opengrok.indexer.configuration.RuntimeEnvironment;
+import org.opengrok.indexer.index.Indexer;
+import org.opengrok.indexer.logger.LoggerFactory;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+/**
+ * Encapsulates logic of meter registry setup and handling.
+ * Generally, the web application publishes metrics to Prometheus and the Indexer to StatsD.
+ */
 public final class Metrics {
 
-    private static final PrometheusMeterRegistry registry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
+    private static final Logger LOGGER = LoggerFactory.getLogger(Metrics.class);
+
+    private static final StatsdConfig statsdConfig = new StatsdConfig() {
+        @Override
+        public String get(String k) {
+            return null;
+        }
+
+        @Override
+        public StatsdFlavor flavor() {
+            String flavor = RuntimeEnvironment.getInstance().getStatsdConfig().getFlavor().toLowerCase();
+            switch (flavor) {
+                case "etsy":
+                    return StatsdFlavor.ETSY;
+                case "datadog":
+                    return StatsdFlavor.DATADOG;
+                case "sysdig":
+                    return StatsdFlavor.SYSDIG;
+                case "telegraf":
+                    return StatsdFlavor.TELEGRAF;
+            }
+
+            return null;
+        }
+
+        @Override
+        public int port() {
+            return RuntimeEnvironment.getInstance().getStatsdConfig().getPort();
+        }
+
+        @Override
+        public String host() {
+            return RuntimeEnvironment.getInstance().getStatsdConfig().getHost();
+        }
+
+        @Override
+        public boolean buffered() {
+            return true;
+        }
+    };
+
+    private static PrometheusMeterRegistry prometheusRegistry;
+    private static StatsdMeterRegistry statsdRegistry;
 
     static {
+        MeterRegistry registry;
+
+        if (RuntimeEnvironment.getInstance().getStatsdConfig().isEnabled()) {
```

Comment:
Why not use `RuntimeEnvironment.getInstance().isIndexer()` here as well?

Reply:
because statsd does not have to be configured. I don't want statsd export to be enabled by default.